File: app/src/main/java/forpdateam/ru/forpda/presentation/forum/ForumView.java
package forpdateam.ru.forpda.presentation.forum;

import forpdateam.ru.forpda.api.forum.models.ForumItemTree;
import forpdateam.ru.forpda.common.mvp.IBaseView;

/**
 * Created by radiationx on 03.01.18.
 */

public interface ForumView extends IBaseView {

    void showForums(ForumItemTree forumRoot);
}

File: app/src/main/java/forpdateam/ru/forpda/presentation/forum/ForumView.kt
package forpdateam.ru.forpda.presentation.forum

import forpdateam.ru.forpda.common.mvp.IBaseView
import forpdateam.ru.forpda.entity.remote.forum.ForumItemFlat
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.viewstate.strategy.SkipStrategy
import moxy.viewstate.strategy.StateStrategyType
import ru.radiationx.coretypes.ForumId

/**
 * Created by radiationx on 03.01.18.
 */

@StateStrategyType(AddToEndSingleStrategy::class)
interface ForumView : IBaseView {
    fun showForums(forums: List<ForumItemFlat>)

    fun scrollToForum(forumId: ForumId)

    @StateStrategyType(SkipStrategy::class)
    fun onMarkRead()

    @StateStrategyType(SkipStrategy::class)
    fun onMarkAllRead()

    @StateStrategyType(SkipStrategy::class)
    fun onAddToFavorite(result: Boolean)
}
